pkgs/development/ruby-modules/gem/default.nix

```diff
@@ -41,7 +41,6 @@ lib.makeOverridable (
 , patches ? []
 , gemPath ? []
 , dontStrip ? true
-, remotes ? ["https://rubygems.org"]
```

Comment:
The main reason I moved this under `source` is because `remotes` is only useful for some types of sources, e.g. `url` and `gem` but not `git`, and so I wanted to keep this nested under `source` like the other source-dependent attributes.

Also, I didn't see any places where this is used with a non-default value in Nixpkgs; I fixed the one instance where it was set. I don't think there's a large backwards-compatibility burdnen, as bundix always passes the `remotes` value under `source`, so only manual calls to `buildRubyGem` would be affected.